packages/hardhat-core/src/builtin-tasks/compile.ts

```diff
@@ -953,17 +962,25 @@ subtask(TASK_COMPILE_SOLIDITY_COMPILE_JOB)
 
       await run(TASK_COMPILE_SOLIDITY_CHECK_ERRORS, { output, quiet });
 
-      const { artifactsEmittedPerFile } = await run(
-        TASK_COMPILE_SOLIDITY_EMIT_ARTIFACTS,
-        {
-          compilationJob,
-          input,
-          output,
-          solcBuild,
-        }
-      );
+      const artifactsEmittedPerFile =
```

Comment:
I try to avoid complex ternaries, I found them harder to read. Here I would do:

```
let artifactsEmittedPerFile = [];
if (emitsArtifacts) {
  artifactsEmittedPerFile = await run(...);
}
```